language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -510,4 +514,30 @@ public void didClose(DidCloseTextDocumentParams params) {
     @Override
     public void didSave(DidSaveTextDocumentParams params) {
     }
+
+    @JsonRequest
+    public CompletableFuture<List<FoldingRange>> foldingRange(FoldingRangeRequestParams params) {
+        return CompletableFuture.supplyAsync(() -> {
+            List<FoldingRange> emptyFoldingRangeList = new ArrayList<>();
```

Comment:
Let's remove this variable and when we need to return an empty list, use `Collections.emptyList()` API

Reply:
Updated.